### PR TITLE
Shereen - Unit tests for EditConfirmModal component within UserProfileModal

### PR DESCRIPTION
**Pull request by pshereen**
# Description
Unit tests for EditConfirmModal component

## How to test:
1. check into current branch
2. do `npm install` and `npm test EditConfirmModal.test.js` to run this PR locally
3. check all 4 tests are passed

## Screenshots or videos of changes:

image

### CODE REVIEW

**Review by ShengjieMao — DISMISSED** [review]
All tests passed. Props are well considered and tested, errors and edges cases also included in the test.

**Review by yaow62023 — DISMISSED** [review]
Hi, All the test are passed. 
Screenshot 2024-03-02 at 9 55 21 AM

**Comment by nnamdixobi** [comment]
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/80783443/265bb9e7-0b75-4bf9-aa78-3b0e12e7195b

Tested as instructed and all tests passed successfully

File: src/components/UserProfile/UserProfileModal/__tests__/EditConfirmModal.test.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { screen, render, fireEvent } from '@testing-library/react';
+import { userProfileMock } from '../../../../__tests__/mockStates';
+import EditConfirmModal from '../EditConfirmModal';
+
+
+function sleep(ms) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
```
Comment (FyisFe) [inline src/components/UserProfile/UserProfileModal/__tests__/EditConfirmModal.test.js:9]: Please consider removing this function, it appears to be unused.

**Review by FyisFe — CHANGES_REQUESTED** [review]
All tests passed.

image

**Review by KurtisIvey — CHANGES_REQUESTED** [review]
Requesting changes due to lack of coverage. There definitely needs to be more than 2 tests here. Your sleep function also serves no purpose.

- test when the modal is closed to ensure its not rendered
- test to confirm functionality of modalmessage
- test confirm functionality of modal title
- test when modalTitle or modalMessage is empty to ensure proper functionality
- test when useProfile set to null to ensure it functions as intended or if it shouldn't work when set to null

To all the other reviewers, 
    Check for test coverage, and not just whether his tests pass lol.

**Review by YHX6 — DISMISSED** [review]
All the test cases works correctly!
Screenshot 2024-03-02 at 18 15 37

**Review by AaronPersaud — DISMISSED** [review]
Hi @pshereen, I have tested and everything has passed! Well done.

image

**Review by meetpadhiar4 — DISMISSED** [review]
Hi, I reviewed your PR and it passed all the test cases. Please refer the screenshot attached.
PR 2014

**Review by Pandani07 — DISMISSED** [review]
This PR works as intended, all the test cases passed

Screenshot 2024-03-04 at 9 24 28 PM

File: src/components/UserProfile/UserProfileModal/__tests__/EditConfirmModal.test.js (hunk 1 above)
Comment (pshereen) [reply src/components/UserProfile/UserProfileModal/__tests__/EditConfirmModal.test.js:9]: Thak you for pointing it out.
Was added to test a different feature.
Will remove it.

**Comment by dhairya-mehta** [comment]
Hey, I verified all the 4 test cases!
image

**Review by vikrambadhan — APPROVED** [review]
Hi, I tested your PR and all the test cases passed!

Week 28 - PR2

**Comment by one-community** [comment]
Thank you all, merging!

**Pull request merged**